File: SoftUni - Advanced/Defining Classes/Defining Classes - Exercise/06. Speed Racing/Car.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace _6.Speed_Racing
{
    public class Car
    {
        public string  Model { get; set; }

        public double FuelAmount { get; set; }

        public double FuelConsumptionPerKilometer { get; set; }

        public double TravelledDistance { get; set; }

        public bool Drive(double distanceTraveled)
        {
            double neededFuel = distanceTraveled 
                                * this.FuelConsumptionPerKilometer;
            if (neededFuel >this .FuelAmount)
            {
                return false;
            }

            this.FuelAmount -= neededFuel;
            this.TravelledDistance += distanceTraveled;
            return true;
        }
    }
}

File: SoftUni - Java Advanced/Defining Classes/Define Classes - Exercise/_03_Speed_Racing/Car.java
package exercise._03_Speed_Racing;

public class Car {
    private String model;
    private double fuelAmount;
    private double fuelCostPerKm;
    private int traveledDistance;

    public Car(String model, double fuelAmount, double fuelCostPerKm) {
        this.model = model;
        this.fuelAmount = fuelAmount;
        this.fuelCostPerKm = fuelCostPerKm;
        this.traveledDistance = 0;
    }

    public boolean canMove(int distance) {
        return this.fuelAmount >= calculateNeededFuel(distance);
    }
    public void increaseDistance(int distance) {
        this.traveledDistance += distance;
    }
    public void decreaseFuel(int distance) {
        double neededFuel = calculateNeededFuel(distance);
        this.fuelAmount -= neededFuel;
    }

    private double calculateNeededFuel(double distance) {
        return this.fuelCostPerKm * distance;
    }

    public String toString(){
        return String.format("%s %.2f %d", this.model, this.fuelAmount, this.traveledDistance);
    }

}
